airbyte-cdk/python/airbyte_cdk/sources/streams/core.py

```diff
@@ -35,6 +35,11 @@
 # value is used to indicate that stream should not load the incoming state value
 FULL_REFRESH_SENTINEL_STATE_KEY = "__ab_full_refresh_state_message"
 
+# Streams that don't have cursors should emit this
+# We still want to emit a state message at the end of the stream
+# So that we can return record counts, but we aren't returning a cursor
+NO_CURSOR_SENTINEL_STATE_KEY = "__ab_no_cursor_state_message"
```

Comment:
Just a heads up that as the temporary workaround we skipped setting state when we saw the old sentinel value here:
https://github.com/airbytehq/airbyte/blob/e74d9363503c6672723ee8db452df824f51e0d9d/airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py#L221-L223

Once the platform starts deciding on when to pass state then that logic can get removed and we always trust incoming state

Reply:
heck yeah! [59916f9](https://github.com/airbytehq/airbyte/pull/36999/commits/59916f9c0025d165374407860fcd142714dd63f4)